test/test_util.py

```diff
@@ -95,6 +96,8 @@ def test_invalid_host(self):
             'http://google.com:foo',
             'http://::1/',
             'http://::1:80/',
+            'http://google.com:-80',
+            six.u('http://google.com:\xb2\xb2'),  # \xb2 = ^2
```

Comment:
Sure, some retry-after checks could be added to test_util. But before doing that I'd like to know your thoughts on the exact behavior it should have regarding input such as negative integers, positive with + prefix, and surrounding whitespace (see my initial comment in this PR).